pkg/acsengine/defaults.go

```diff
@@ -102,15 +102,17 @@ func setDefaultCerts(a *api.Properties) (bool, error) {
 		ips = append(ips, net.IP{firstMasterIP[12], firstMasterIP[13], firstMasterIP[14], firstMasterIP[15] + byte(i)})
 	}
 
```

Comment:
I think this file and the one below could be made more straightforward.

2. TryParse the ca cert and private key here.

3. Pass it as a `*pkiKeyCertPair` into `CreatePki` (much more clear than strings that might be empty) and then `CreatePki` has simpler logic for determining whether to add it's own or not.

This has the nice benefit of removing this code, a little bit lower, which was confusing for a bit:

```diff
+ if len(a.CertificateProfile.CaCertificate) == 0 || len(a.CertificateProfile.GetCAPrivateKey()) == 0 {
+		a.CertificateProfile.CaCertificate = caPair.CertificatePem
+		a.CertificateProfile.SetCAPrivateKey(caPair.PrivateKeyPem)
+	}
```